src/cli/trn_cli_common.c

```diff
@@ -38,112 +38,156 @@ cJSON *trn_cli_parse_json(const char *buf)
 	return conf_json;
 }
 
-int trn_cli_parse_xdp(const cJSON *jsonobj, rpc_trn_xdp_intf_t *xdp_intf)
+int trn_cli_parse_json_string(const cJSON *jsonobj, const char *const key, char *buf)
 {
-	cJSON *xdp_path = cJSON_GetObjectItem(jsonobj, "xdp_path");
-	cJSON *pcapfile = cJSON_GetObjectItem(jsonobj, "pcapfile");
+	cJSON *item = cJSON_GetObjectItem(jsonobj, key);
 
-	if (xdp_path == NULL) {
-		print_err("Missing path for xdp program to load.\n");
+	if (item == NULL) {
+		print_err("Missing %s\n", key);
+		return -EINVAL;
+	} else if (!cJSON_IsString(item)) {
+		print_err("Invalid %s type, should be string\n", key);
 		return -EINVAL;
-	} else if (cJSON_IsString(xdp_path)) {
-		strcpy(xdp_intf->xdp_path, xdp_path->valuestring);
 	}
+	strcpy(buf, item->valuestring);
+	return 0;
+}
+
+int trn_cli_parse_json_number_ip(const cJSON *jsonobj, const char *const key,
+	unsigned int *buf)
+{
+	cJSON *item = cJSON_GetObjectItem(jsonobj, key);
 
-	if (pcapfile == NULL) {
-		xdp_intf->pcapfile = NULL;
-		print_err(
-			"Warning: Missing pcapfile. Packet capture will not be available for the interface.\n");
-	} else if (cJSON_IsString(pcapfile)) {
-		strcpy(xdp_intf->pcapfile, pcapfile->valuestring);
+	if (item == NULL) {
+		print_err("Missing %s\n", key);
+		return -EINVAL;
+	} else if (!cJSON_IsNumber(item)) {
+		print_err("Invalid %s type, should be number\n", key);
+		return -EINVAL;
 	}
+	*buf = (unsigned int)item->valuedouble;
 	return 0;
 }
 
-int trn_cli_parse_zeta_key(const cJSON *jsonobj,
-			   struct rpc_trn_zeta_key_t *zeta_key)
+int trn_cli_parse_json_number_mac(const cJSON *jsonobj, const char *const key,
+	unsigned char *buf)
 {
-	cJSON *id = cJSON_GetObjectItem(jsonobj, "id");
+	cJSON *item = cJSON_GetObjectItem(jsonobj, key);
 
-	if (id == NULL) {
-		print_err("Error: Zeta ID Error\n");
+	if (item == NULL) {
+		print_err("Missing %s\n", key);
 		return -EINVAL;
-	} else if (cJSON_IsString(id)) {
-		zeta_key->id = atoi(id->valuestring);
-	} else {
-		print_err("Error: ID Error\n");
+	} else if (!cJSON_IsNumber(item)) {
+		print_err("Invalid %s type, should be number\n", key);
 		return -EINVAL;
 	}
+	uint64_t vd = ((uint64_t)item->valuedouble & 0xFFFFFFFFFFFF);
```

Comment:
Is this a secret trick 😯
Doesn't & with 0xFFFFFFFFFFFF just give you back the same number?

Reply:
That is not 64bit mask as only lower 48bits are MAC